GitUI/CommandsDialogs/RememberFileContextMenuController.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using GitCommands;
+using GitUI.UserControls;
+using GitUIPluginInterfaces;
+using JetBrains.Annotations;
+
+namespace GitUI.CommandsDialogs
+{
+    public class RememberFileContextMenuController
+    {
+        // Singleton accessor
+        public static RememberFileContextMenuController Default { get; } = new RememberFileContextMenuController();
+
+        /// <summary>
+        /// The remembered file status item, to diff with other files and commits
+        /// </summary>
+        public FileStatusItem RememberedDiffFileItem { get; set; }
+
+        // Note that the methods in this class are without side effects (static)
+
+        /// <summary>
+        /// Get a FileStatusItem that can be used to compare with
+        /// </summary>
+        /// <param name="name">The file name</param>
+        /// <param name="rev">The Git revision</param>
+        /// <returns>The FileStatusItem</returns>
+        public FileStatusItem getFileStatusItem(string name, GitRevision rev)
```

Comment:
```suggestion
        public static FileStatusItem CreateFileStatusItem(string name, GitRevision rev)
```
Though I am not 100% sure regarding `Create...` vs. `Get...`.

Reply:
All methods in this class is static, but @RussKie prefers them to be instanced.
Will change the name though (As you wrote Create first, probably be that.)

Reply:
So label it with the `[Pure]` [attribute](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.contracts.pureattribute?view=netcore-3.1), please.